cli/compile/compile.go

```diff
@@ -112,7 +112,7 @@ func NewCommand() *cobra.Command {
 	command.RegisterFlagCompletionFunc("programmer", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
 		return arguments.GetInstalledProgrammers(), cobra.ShellCompDirectiveDefault
 	})
-	command.Flags().BoolVar(&compilationDatabaseOnly, "only-compilation-database", false, tr("Just produce the compilation database, without actually compiling."))
+	command.Flags().BoolVar(&compilationDatabaseOnly, "only-compilation-database", false, tr("Just produce the compilation database, without actually compiling. All build commands are skipped except all the pre-* hooks."))
```

Comment:
```suggestion
	command.Flags().BoolVar(&compilationDatabaseOnly, "only-compilation-database", false, tr("Just produce the compilation database, without actually compiling. All build commands are skipped except pre* hooks."))
```
- Make more concise
- Remove the hyphen from hook pattern to match [the property names](https://arduino.github.io/arduino-cli/dev/platform-specification/#pre-and-post-build-hooks-since-arduino-ide-165)